### PR TITLE
Resolve default values of string parameters as downstream job. This i…

### PR DESCRIPTION
**Pull request by J-cztery**
…s helpfull in configurations where inheritance plugin is used because parameters are also means to parametrize jobs and not only builds.

### CODE REVIEW

**Comment by J-cztery** [comment]
This partially addresses [JENKINS-11280]

File: src/main/java/hudson/plugins/parameterizedtrigger/BuildTrigger.java [hunk 1]
```diff
@@ -87,14 +92,33 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
...
 	@Override
 	public void buildDependencyGraph(AbstractProject owner, DependencyGraph graph) {
 		// Can only add dependencies in Hudson 1.341 or higher
 		if (!canDeclare(owner)) return;
 
+        //get default parameter values as environment. 
+		EnvVars envVars = GetDefaultParameterValuesAsEnvVars(owner);
 		for (BuildTriggerConfig config : configs) {
-			List<AbstractProject> projectList = config.getProjectList(owner.getParent(), null);
+			List<AbstractProject> projectList = config.getProjectList(owner.getParent(),envVars);
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/parameterizedtrigger/BuildTrigger.java]: It's not the only place where such construct is being used

**Comment by oleg-nenashev** [comment]
The PR needs more investigation. IIRC @recena did the same thing in SVN plugin, so maybe he could share his experience.

**Comment by recena** [comment]
@oleg-nenashev It was a little bit different but a very similar need. Anyway, this PR requires more investigation and review PR title, description and coding style.

**Comment by J-cztery** [comment]
Guys, could you please be precise on what do you want me to do on this one to get it merged?

**Comment by J-cztery** [comment]
@svanoort Are you working on this PR? @recena Could you please do that then?

**Comment by J-cztery** [comment]
@svanoort , @recena Any news on this one?

**Comment by J-cztery** [comment]
@oleg-nenashev could you please help me understand what can i do to get this request merged?

**Comment by daniel-beck** [comment]
No tests make it difficult to determine what this does. So is the missing documentation.

**Comment by J-cztery** [comment]
By documentation, do you mean Readme? Or something else?
Ok, I will add a test. 
Original problem statement is here: JENKINS-11280. The changes are quite limited so i did not realized there might be problems with understanding what problems i am trying to solve here.

**Pull request closed without merging**